channeldb/peers.go

```diff
@@ -0,0 +1,158 @@
+package channeldb
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/lightningnetwork/lnd/channeldb/kvdb"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+var (
+	// peerBucket is the name of a top level bucket in which we store
+	// information about our peers. Peers are keyed by their pubkey, with
+	// data nested per-channel point.
+	//
+	//
+	// peers-bucket
+	//      |
+	//      |-- <peer-pubkey>
+	//      |        |
+	//      |        |--<chan-point>
+	//     	|        |	|--flap-count-key: <flap count>
+	//      |        |--<chan-point>
+	//     	|		|--flap-count-key: <flap count>
+	//      |
+	//      |-- <peer-pubkey>
+	//      |        |--<chan-point>
```

Comment:
Something I don't understand is why do we store the flap count for each channel? Isn't it supposedly the same?

Reply:
Ah I think I was unclear in my review comment a ways above. Perhaps a better architecture would be a sub-bucket under the peer bucket that stores the total count and then each chan point has its individual flap count. Reason being that granularity will help with banning (per-peer OR per-channel): https://github.com/lightningnetwork/lnd/issues/1183